common/flatpak-dir.c

```diff
@@ -8080,6 +8267,168 @@ flatpak_dir_remote_fetch_summary (FlatpakDir   *self,
   return TRUE;
 }
 
+static FlatpakRemoteState *
+_flatpak_dir_get_remote_state (FlatpakDir   *self,
+                               const char   *remote,
+                               gboolean      optional,
```

Comment:
What is `optional` used for? Are there any invariants to enforce, like `(collection_id != NULL) ⇒ optional`?

Reply:
Optional means we don't need the entire state, load as much as possible. In particular, this is to handle the p2p offline case.